### ISSUE TITLE
External SST::Statistics libraries do not have access to core simulation object

### ISSUE THREAD

**Issue by jleidel** [sstsimulator/sst-core#1058]
**Is your feature request related to a problem? Please describe.**
SST provides the ability to construct external libraries that provide `SST::Statistics` mechanisms to store simulation telemetry to orthogonal storage mediums.  This is analogous to the console and CSV statistics interfaces.  Currently, external statistics libraries cannot access the core simulation object.  For example, the following code in the CSV statistics interface cannot be used in externally developed statistics libraries: 

https://github.com/sstsimulator/sst-core/blob/master/src/sst/core/statapi/statoutputcsv.cc#L92C14-L92C46

**Describe the solution you'd like**
We would like to request an API that permits external statistics libraries to retrieve and/or query the core simulation object in order to determine the number of ranks (et al. degree of parallelism).  Other bits of info might be interesting as well.  

**Describe alternatives you've considered**
We've attempted to get around this for MPI-centric simulations by retrieving the number of ranks from the MPI_COMM_WORLD communicator object.  This is not ideal as we really shouldn't need to access MPI directly from external libraries.  

**Additional context**
No other additional context required.

**Comment by gvoskuilen** [comment]
This is a useful addition and should be relatively straightforward to support. I'm going to mark "in progress" to denote that we're looking into enabling this.

**Comment by jleidel** [comment]
Did this ever make it into the installed API?

**Comment by gvoskuilen** [comment]
Yes, additional functions to access core simulation state including getRank/getNumRanks were added in #1081. I think we can close this issue unless there are additional missing API calls.
